e2e/wdio/wdio.conf.js

```diff
@@ -0,0 +1,337 @@
+  // onWorkerStart: function (cid, caps, specs, args, execArgv) {
+  // },
+  /**
+   * Gets executed just before initialising the webdriver session and test framework. It allows you
+   * to manipulate configurations depending on the capability or spec.
+   * @param {Object} config wdio configuration object
+   * @param {Array.<Object>} capabilities list of capabilities details
+   * @param {Array.<string>} specs List of spec file paths that are to be run
+   */
+  // beforeSession: function (config, capabilities, specs) {
+  // },
+  /**
+   * Gets executed before test execution begins. At this point you can access to all global
+   * variables like `browser`. It is the perfect place to define custom commands.
+   * @param {Array.<Object>} capabilities list of capabilities details
+   * @param {Array.<string>} specs        List of spec file paths that are to be run
+   * @param {Object}         browser      instance of created browser/device session
+   */
+  before: async function(capabilities, specs) {
+    // require('expect-webdriverio')
+    global.assert = require('assert');
+    await Object.keys(commands).forEach((key) => {
+      browser.addCommand(key, commands[key]);
+    });
+  },
+  /**
+   * Runs before a WebdriverIO command gets executed.
+   * @param {string} commandName hook command name
+   * @param {Array} args arguments that command would receive
+   */
+  // beforeCommand: function (commandName, args) {
+  // },
+  /**
+   * Hook that gets executed before the suite starts
+   * @param {Object} suite suite details
+   */
+  // beforeSuite: function (suite) {
+  // },
+  /**
+   * Function to be executed before a test (in Mocha/Jasmine) starts.
+   */
+  // beforeTest: function (test, context) {
+  // },
+  /**
+   * Hook that gets executed _before_ a hook within the suite starts (e.g. runs before calling
+   * beforeEach in Mocha)
+   */
+  // beforeHook: function (test, context) {
+  // },
+  /**
+   * Hook that gets executed _after_ a hook within the suite starts (e.g. runs after calling
+   * afterEach in Mocha)
+   */
+  // afterHook: function (test, context, { error, result, duration, passed, retries }) {
+  // },
+  /**
+   * Function to be executed after a test (in Mocha/Jasmine).
+   * @param test
+   * @param context
+   * @param root0
+   * @param root0.error
+   * @param root0.result
+   * @param root0.duration
+   * @param root0.passed
+   * @param root0.retries
+   */
+  afterTest: function(test, context, { error, result, duration, passed, retries }) {
+    const resultsDir = `./results/screenshots/${test.fullName.split('*')[0]}`;
+    if (error) {
+      if (!fs.existsSync(resultsDir)) {
+        fs.mkdirSync(resultsDir);
+      }
+      const attachmentPath = `${resultsDir}/${test.description}.png`;
+      browser.saveScreenshot(attachmentPath);
+    }
+  },
+
+  /**
+   * Hook that gets executed after the suite has ended
+   * @param {Object} suite suite details
+   */
+  // afterSuite: function (suite) {
```

Comment:
A bunch of these hooks are commented out - should we remove or any harm leaving uncommented? Guessing these are part of the wdio boilerplate.